packages/amplify-provider-awscloudformation/src/root-stack-builder/root-stack-builder.ts

```diff
@@ -84,21 +68,19 @@ export class AmplifyRootStack extends cdk.Stack implements AmplifyRootStackTempl
    * @param logicalId
    */
   addCfnParameter(props: cdk.CfnParameterProps, logicalId: string): void {
-    try {
-      if (this._cfnParameterMap.has(logicalId)) {
-        throw new Error('logical Id already Exists');
-      }
-      this._cfnParameterMap.set(logicalId, new cdk.CfnParameter(this, logicalId, props));
-    } catch (error) {
-      throw new Error(error);
+    if (this._cfnParameterMap.has(logicalId)) {
+      throw new AmplifyError('DuplicateLogicalIdError', {
+        message: `Logical Id already exists: ${logicalId}.`,
+        link: `${AMPLIFY_SUPPORT_DOCS.CLI_PROJECT_TROUBLESHOOTING.url}`,
```

Comment:
this seems to be repeated a bunch of times. Does it make sense to have this link be inserted by default by the error handler if a link is not provided? Or alternatively, have an error factory function like `amplifyErrorWithTroubleshootingLink(type: ErrorType, options: Options)` which will construct an error with the given info plus the default troubleshooting link.

Also, here and everywhere else, it can just be written as:
```suggestion
  link: AMPLIFY_SUPPORT_DOCS.CLI_PROJECT_TROUBLESHOOTING.url,
```